pkg/remote/client.go

```diff
@@ -117,7 +117,7 @@ func newWriteRequestBody(series []*prompb.TimeSeries) ([]byte, error) {
 	}
 	if snappy.MaxEncodedLen(len(b)) < 0 {
 		return nil, fmt.Errorf("the protobuf message is too large to be handled by Snappy encoder; "+
-			"size: %d, limit: %d", len(b), 0xffffffff)
+			"size: %d, limit: %d", len(b), uint(0xffffffff))
```

Comment:
It sounds to me the issue is more running on a 32-bit platform than being ARM. Am I missing something?

If my assumption is correct than doing the cast would generate a bad error message because the value is different based on the bit size.

Something like the following should be more accurate:

```
maxLen = math.MaxInt32
if bits.UintSize == 64 {
  maxLen = 0xffffffff
}
```

Reply:
ah, you are right, nice catch